arraycontext/impl/pyopencl/fake_numpy.py

```diff
@@ -208,6 +210,31 @@ def all(self, a):
             result = result.get()[()]
         return result
 
+    def array_equal(self, a, b):
+        def as_scalar(bool_value):
```

Comment:
This seems to unnecessarily round-trip to the device. Why not just return the (known) `bool` result?